### PR TITLE
Implement the autograder skeleton

### PR DESCRIPTION
**Pull request by lowjoel**

### CODE REVIEW

**Comment by lowjoel** [comment]
@allenwq Have a look please. See if the design makes sense to you.

File: app/services/course/assessment/answer/auto_grading_service.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+class Course::Assessment::Answer::AutoGradingService
+  class << self
+    # Picks the grader for the given grading, then grades into the given
+    # +Course::Assessment::Answer::AutoGrading+ object.
+    #
+    # @param [Course::Assessment::Answer::AutoGrading] auto_grading The object to store grading
```
Comment (allenwq) [inline app/services/course/assessment/answer/auto_grading_service.rb]: I think it's not a good idea to use input as well as output
And how about change the params to `answer` ? I think `grade(answer)` makes more sense. 

My two cents, what do you think ? @lowjoel

Comment (lowjoel) [inline app/services/course/assessment/answer/auto_grading_service.rb]: I currently enforce this to be one answer one grading. I'm not sure if there's a need to have one answer graded multiple times.

In any case, this should not be directly called by the client code. I'll abstract this behind the answer model so the answer model handles the job.

Comment (allenwq) [inline app/services/course/assessment/answer/auto_grading_service.rb]: OK, then it's fine

**Pull request merged**